sdk/src/org.graalvm.nativeimage/src/org/graalvm/nativeimage/impl/ReflectionRegistry.java

```diff
@@ -46,7 +46,12 @@
 
 public interface ReflectionRegistry {
     default void register(ConfigurationCondition condition, Class<?>... classes) {
-        Arrays.stream(classes).forEach(clazz -> register(condition, false, clazz));
+        Arrays.stream(classes).forEach(clazz -> {
+            if (clazz == null) {
+                throw new NullPointerException("Trying to register null for reflection");
```

Comment:
I'm no native speaker but "trying" may not be clear enough for users here. How about:

```suggestion
                throw new NullPointerException("Cannot register null value as class for reflection. Please ensure that all values you register are not null.");
```

Reply:
I updated all messages following this pattern.